src/tools/error-codes/main.rs

```diff
@@ -0,0 +1,57 @@
+use std::env::args;
+use std::fs::read_dir;
+use std::path::Path;
+use std::process::{exit, Command};
+
+fn get_error_codes<P: AsRef<Path>>(folder_path: P) -> Vec<String> {
+    let mut ret = Vec::with_capacity(10);
```

Comment:
Please drop the capacity here or add a comment as to why 10. AFAICT, it's not true that we expect exactly 10 elements, and regardless we shouldn't really be optimizing this code.